auth/src/main/java/org/aerogear/auth/IKeycloakUserPrincipal.java

```diff
@@ -0,0 +1,16 @@
+package org.aerogear.auth;
+
+
+import org.aerogear.auth.credentials.OIDCCredentials;
+
+public interface IKeycloakUserPrincipal extends IUserPrincipal {
```

Comment:
Those tokens are not specific to Keycloak, they are all part of the OIDC protocol, so I don't think we need a separate interface for it.